Grbl_Esp32/src/Exec.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <map>
+
+// System executor bit map. Used internally by realtime protocol as realtime command flags,
+// which notifies the main program to execute the specified realtime command asynchronously.
+// NOTE: The system executor uses an unsigned 8-bit volatile variable (8 flag limit.) The default
+// flags are always false, so the realtime protocol only needs to check for a non-zero value to
+// know when there is a realtime command to execute.
+struct ExecStateBits {
+    uint8_t statusReport : 1;
+    uint8_t cycleStart : 1;
+    uint8_t cycleStop : 1;  // Unused, per cycle_stop variable
+    uint8_t feedHold : 1;
+    uint8_t reset : 1;
+    uint8_t safetyDoor : 1;
+    uint8_t motionCancel : 1;
+    uint8_t sleep : 1;
+};
+
+union ExecState {
+    uint8_t       value;
+    ExecStateBits bit;
+};
```

Comment:
You probably have a good reason for this union. 

I'm not sure if ExecStateBits now evaluates to a single byte in all cases. Fortunately, you can enforce it:

static_assert(sizeof(ExecStateBits) == sizeof(uint8_t), "ExecStateBits is not an uint8");

Reply:
The reason is because it was too hard to change the existing code to do it the right way.  So I gave up and did what I could.

I added the assertion.